node/core/backing/src/lib.rs

```diff
@@ -578,6 +593,11 @@ impl CandidateBackingJob {
 			return Ok(());
 		}
 
+		let candidate_hash = candidate.hash();
+		self.add_unbacked_span(&parent_span, candidate_hash);
+		let span = self.get_unbacked_validation_child(&candidate_hash)
+			.expect("just added unbacked span; qed");
```

Comment:
maybe we can avoid this `expect` by returning a reference to span in `add_unbacked_span`?

Reply:
That will duplicate the validation span creation logic which seems even more error-prone.

Reply:
There's room for improvement in the API but gets complicated with `&'a mut self` borrow in `add_unbacked_span`. Didn't want to spend time on it, bigger fish to fry

Reply:
We could use a constant to avoid duplication, we could even merge it into one function.